support-frontend/assets/helpers/customHooks/useFormValidation.ts

```diff
@@ -1,28 +1,32 @@
 import { useCallback, useEffect, useState } from 'react';
 import { validateForm } from 'helpers/redux/checkout/checkoutActions';
-import { contributionsFormHasErrors } from 'helpers/redux/checkout/checkoutSelectors';
+import { contributionsFormHasErrors } from 'helpers/redux/selectors/formValidation';
 import {
 	useContributionsDispatch,
 	useContributionsSelector,
 } from 'helpers/redux/storeHooks';
 import { paymentWaiting } from 'pages/contributions-landing/contributionsLandingActions';
 
+type PreventableEvent = {
+	preventDefault: () => void;
+};
+
 /**
  * A hook to wrap any payment handler function.
  * Validates the form, and will only run the handler if the form is valid.
  */
-export function useFormValidation(
-	paymentHandler: (event: React.MouseEvent<HTMLButtonElement>) => void,
-): (event: React.MouseEvent<HTMLButtonElement>) => void {
-	const [clickEvent, setClickEvent] =
-		useState<React.MouseEvent<HTMLButtonElement> | null>(null);
+export function useFormValidation<
+	EventType extends PreventableEvent = React.MouseEvent<HTMLButtonElement>,
+>(paymentHandler: (event: EventType) => void): (event: EventType) => void {
```

Comment:
I had to change the type signature here to handle Stripe's `StripePaymentRequestButtonElementClickEvent`, which for some reason does not inherit from `Event` or anything else. Most of the time we'll just default to the React button click event.